crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -56,14 +56,22 @@ pub struct AuthorityPerpetualTables {
     /// a digest of ObjectDigest::deleted(), along with a link to the transaction that deleted it.
     pub(crate) parent_sync: DBMap<ObjectRef, TransactionDigest>,
 
-    /// A map between the transaction digest of a certificate that was successfully processed
-    /// (ie in `certificates`) and the effects its execution has on the authority state. This
-    /// structure is used to ensure we do not double process a certificate, and that we can return
-    /// the same response for any call after the first (ie. make certificate processing idempotent).
+    /// A map between the transaction digest of a certificate to the effects of its execution.
+    /// We store effects into this table in two different cases:
+    /// 1. When a transaction is synced through state_sync, we store the effects here. These effects
+    /// are known to be final in the network, but may not have been executed locally yet.
+    /// 2. When the transaction is executed locally on this node, we store the effects here. This means that
+    /// it's possible to store the same effects twice (once for the synced transaction, and once for the executed).
```

Comment:
Do we want to add logic (if we haven't already) to ensure that if we do write to the db and there is already an entry that it matches whats already there?

Reply:
Good idea. Added a TODO. At this point I am hoping to not grow this PR and let more people to take a look.